FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/model/LoginModel.java

```diff
@@ -1,23 +1,42 @@
 package com.pps.globant.fittracker.mvp.model;
 
-import com.facebook.CallbackManager;
+import android.app.Activity;
+
+import com.pps.globant.fittracker.model.User;
 import com.pps.globant.fittracker.utils.FacebookLoginProvider;
-import com.squareup.otto.Bus;
 
 public class LoginModel {
-    private final Bus bus;
-    private final CallbackManager callbackManager;
+    private FacebookLoginProvider facebookLoginProvider;
+    private User activeUser;
 
-    public LoginModel(CallbackManager callbackManager, Bus bus) {
-        this.bus = bus;
-        this.callbackManager = callbackManager;
+    public LoginModel(FacebookLoginProvider facebookLoginProvider) {
+        activeUser = null;
+        this.facebookLoginProvider = facebookLoginProvider;
     }
 
-    public void registerFbCallbacks() {
-        FacebookLoginProvider.registerCallback(bus, callbackManager);
+    public void fbLogOut() {
+        facebookLoginProvider.logOut();
     }
 
-    public void fbLogOut() {
-        FacebookLoginProvider.logOut();
+
+    public void fbLogIn(Activity activity) {
+        facebookLoginProvider.logIn(activity);
     }
+
+    public boolean isFbLogedIn() {
+        return facebookLoginProvider.isLoginTokenActive();
+    }
+
+    public User getUser() {
+        return activeUser;
+    }
+
+    public void restoreState() {
+        facebookLoginProvider.restoreState();
+    }
+
+    public void setUser(User user) {
+        this.activeUser=user;
```

Comment:
the reserver word "this" is not neccesary in that case, because the names are different

Reply:
thanks, it is an habit aajja